src/silx/gui/colors.py

```diff
@@ -828,6 +841,8 @@ def _setFromDict(self, dic: dict):
         if name is not None:
             self.setName(name)
         else:
+            # if name is None => colors is not None
+            assert colors is not None
```

Comment:
Best to stop using asserts in code and raise appropriate exception:
```suggestion
            if colors is None:
                raise ValueError("name and colors cannot be both None")
```

Reply:
I disagree, assert is supposed to be an assert. No more no less than what it is in C or C++.

That's an extra check which is not mandatory.

Reply:
Indeed, there is a `raise ValueError` above in the code